lib/roo_on_rails/checks/base.rb

```diff
@@ -84,10 +96,15 @@ def signature
 
       # Run each dependency, then mark them as run.
       def resolve(deps)
-        deps.each do |dep|
-          context.deps ||= {}
-          context.deps[dep.signature.join('/')] ||= dep.run
-        end
+        context.deps ||= {}
+        deps.map { |dep|
+          sig = dep.signature.join('/')
+          if context.deps.key?(sig)
+            context.deps[sig]
+          else
+            context.deps[sig] = dep.run
+          end
+        }.to_a.all?
```

Comment:
I _think_ that in Ruby 2.4 it's required, as `all?` is lazy and we want to evaluate all items (not stop at the first failure).